synapse/metrics/_exposition.py

```diff
@@ -94,7 +109,8 @@ def generate_latest(registry: CollectorRegistry, emit_help: bool = False) -> byt
             # No samples, don't bother.
             continue
 
-        mname = metric.name
+        # Translate to legacy metric name if it has one.
+        mname = LEGACY_METRIC_NAMES.get(metric.name, metric.name)
```

Comment:
I was surprised to see that lookups in LEGACY_METRIC_NAMES wasn't guarded by the new config flag. Is the point that `generate_latest` is only served if we invoke `start_http_server_legacy`?

If so, a comment on generate_latest mentioning that this is old/deprecated/legacy might be handy.